lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -96,7 +96,16 @@ def store(credit_card, options={})
         add_stored_credentials(post, credit_card, options)
         add_recurring_contract(post, options)
         add_address(post, options)
-        commit('authorise', post, options)
+
+        initial_response = commit('authorise', post, options)
+
+        if initial_response.success? == false
+          initial_response
+        elsif initial_response.authorization.split('#')[2].nil? 
```

Comment:
What does this check? Is ```initial_response.authorization.split('#')[2].nil?``` true when this is a recurring transaction? (Looking at ```authorization_from``` linked above, it looked like it would be the opposite -- non-nil when the transaction is recurring -- but I could be interpreting that incorrectly).

Reply:
```
def card_not_successfully_stored?(response)
  response.authorization.split('#')[2].nil?
end
```

Reply:
This has now been refactored to create a new helper method as you suggested above.